src/foam/parse/DateParser.js

```diff
@@ -0,0 +1,727 @@
+          ret = new Date(
+            result.year,
+            result.month,
+            result.day,
+            result.hour || 0,
+            result.minute || 0,
+            result.second || 0,
+            result.millisecond || 0
+          );
+        } else {
+          // Date-only format - use noon GMT (matches DateUtil behavior)
+          ret = new Date(Date.UTC(result.year, result.month, result.day, 12, 0, 0, 0));
+        }
+
+        return this.validateDate(ret, str, result.year, result.month, result.day);
+      }
+    },
+
+    {
+      name: 'parseDateString',
+      documentation: 'Parse a date string - ignores any time component and returns date at noon local time. Returns MAX_DATE for invalid dates.',
+      code: function(str, opt_name) {
+        let result = this.grammar_.parseString(str, opt_name || 'START');
+
+        if ( ! result ) {
+          // Unparseable format - return MAX_DATE
+          return this.validateDate(new Date(NaN), str);
+        }
+
+        // Always return date at noon local time, ignoring time even if present
+        let ret = new Date(result.year, result.month, result.day, 12, 0, 0, 0);
+
+        return this.validateDate(ret, str, result.year, result.month, result.day);
+      }
+    },
+
+    {
+      name: 'parseDateTime',
+      documentation: 'Parse a datetime string using local time - uses time if present, otherwise sets to noon. If timezone is present, converts to UTC. Returns MAX_DATE for invalid dates.',
+      code: function(str, opt_name) {
+        // Trim input to remove leading/trailing whitespace
+        str = str ? str.trim() : str;
+
+        // Use parse() instead of parseString() to get position information
+        this.grammar_.ps.setString(str);
+        let start = this.grammar_.getSymbol(opt_name || 'START');
+        let parseResult = this.grammar_.ps.apply(start, this.grammar_);
+
+        if ( ! parseResult ) {
+          // Unparseable format - return MAX_DATE
+          return this.validateDate(new Date(NaN), str);
+        }
+
+        // Check if entire string was consumed
+        if ( parseResult.pos < str.length ) {
+          // Partial parse - remaining characters indicate invalid format
+          console.warn('DateParser: Partial parse detected. Input:', str, 'Consumed up to position:', parseResult.pos, 'Remaining:', str.substring(parseResult.pos));
+          return this.validateDate(new Date(NaN), str);
+        }
+
+        let result = parseResult.value;
+
+        if ( ! result ) {
+          // Unparseable format - return MAX_DATE
+          return this.validateDate(new Date(NaN), str);
+        }
+
+        // Validate time components if present
+        // Note: Grammar already enforces valid ranges (hour2: 00-23, minute2/second2: 00-59)
+        // but we keep these checks as a safety measure
+        if ( result.hour !== undefined && (result.hour < 0 || result.hour > 23) ) {
+          return this.validateDate(new Date(NaN), str);
```

Comment:
Could you make a Constant for new Date(NaN)?

Reply:
done